tstest/integration: TestC2NPingRequest  starting tailscaled ... text file busy

https://github.com/tailscale/tailscale/actions/runs/14805793639/job/41573762653?pr=15722
```
=== RUN   TestC2NPingRequest
=== PAUSE TestC2NPingRequest
=== CONT  TestC2NPingRequest
    integration.go:527: DERP httpsrv listener: 127.0.0.1:42277
    integration.go:561: control URL: http://127.0.0.1:39331/
    integration.go:834: starting tailscaled: fork/exec /tmp/TestC2NPingRequest1267665213/001/tailscaled: text file busy
    stuntest.go:65: STUN server shutdown
--- FAIL: TestC2NPingRequest (0.10s)
ok	tailscale.com/tstest/jsdeps	0.295s
```

Related to 761aea3036d9ac35fafbd871df8b52c6e2b80cca? But not obvious how if so.

--- a/tstest/integration/integration.go
+++ b/tstest/integration/integration.go
@@ -770,6 +770,27 @@ func (d *Daemon) MustCleanShutdown(t testing.TB) {
 	}
 }
 
+// awaitTailscaledRunnable tries to run `tailscaled --version` until it
+// works. This is an unsatisfying workaround for ETXTBSY we were seeing
+// on GitHub Actions that aren't understood. It's not clear what's holding
+// a writable fd to tailscaled after `go install` completes.
+// See https://github.com/tailscale/tailscale/issues/15868.
+func (n *TestNode) awaitTailscaledRunnable() error {
+	t := n.env.t
+	t.Helper()
+	if err := tstest.WaitFor(10*time.Second, func() error {
+		out, err := exec.Command(n.env.daemon, "--version").CombinedOutput()
+		if err == nil {
+			return nil
+		}
+		t.Logf("error running tailscaled --version: %v, %s", err, out)
+		return err
+	}); err != nil {
+		return fmt.Errorf("gave up trying to run tailscaled: %v", err)
+	}
+	return nil
+}
+
 // StartDaemon starts the node's tailscaled, failing if it fails to start.
 // StartDaemon ensures that the process will exit when the test completes.
 func (n *TestNode) StartDaemon() *Daemon {
@@ -778,6 +799,11 @@ func (n *TestNode) StartDaemon() *Daemon {
 
 func (n *TestNode) StartDaemonAsIPNGOOS(ipnGOOS string) *Daemon {
 	t := n.env.t
+
+	if err := n.awaitTailscaledRunnable(); err != nil {
+		t.Fatalf("awaitTailscaledRunnable: %v", err)
+	}
+
 	cmd := exec.Command(n.env.daemon)
 	cmd.Args = append(cmd.Args,
 		"--statedir="+n.dir,

